s3/s3.go

```diff
@@ -309,69 +318,33 @@ type Key struct {
 //     }
 //
 // See http://goo.gl/YjQTc for details.
-func (b *Bucket) List(prefix, delim, marker string, max int) (result *ListResp, err error) {
-	params := map[string][]string{
-		"prefix":    {prefix},
-		"delimiter": {delim},
-		"marker":    {marker},
-	}
-	if max != 0 {
-		params["max-keys"] = []string{strconv.FormatInt(int64(max), 10)}
-	}
-	req := &request{
-		bucket: b.Name,
-		params: params,
-	}
-	result = &ListResp{}
-	for attempt := attempts.Start(); attempt.Next(); {
-		err = b.S3.query(req, result)
-		if !shouldRetry(err) {
-			break
-		}
-	}
+func (b *Bucket) List(prefix, delim, marker string, max int) (*ListResp, error) {
+
+	req, err := http.NewRequest("GET", b.ResolveS3BucketEndpoint(b.Name), nil)
 	if err != nil {
 		return nil, err
 	}
-	return result, nil
-}
+	req.Close = true
 
-// URL returns a non-signed URL that allows retriving the
```

Comment:
As we discussed:
* This is proposed against v3-unstable, so it will not break anyone.
* SignedURL no longer makes sense as an option since v4 utilizes headers for signing.

I have added the URL function back in.